### ISSUE TITLE
Private Method isn't private

### ISSUE THREAD

**Issue by felipetesc** [marcobambini/gravity#240]
Hello there. I was writing some examples to test and document the access modifiers and I found that the code below works:
```
class TestAccessModifiers{
    public var my_var = "I'm public!";
    private var my_hidden_var = "I'm private!";
    public func get_hidden(){
        return my_hidden_var;
    }
    private func get_public(){
        return my_var;
    }
}
func puts(str){
    System.print(str);
}
func main(){
    var test = TestAccessModifiers();
    test.my_var;
    //puts("Public var holds = " + test.my_var );//works
    //puts("Get hidden var from public method = " + test.get_hidden());//works
    //puts("Private var holds = " + test.my_hidden_var );//works and cause error like it should
    puts("Get public var from private method = " + test.get_public());//doesn't works 

}
```
As result, from the terminal, I get :+1: 
``` Get public var from private method = I'm public! ```
I donno, but I guessing that the keyword private before the keyword func should print the same, or almost the same value, as in  "puts("Private var holds = " + test.my_hidden_var );", which is :+1: RUNTIME ERROR: Unable to find my_hidden_var into class TestAccessModifiers

Thanks!

**Comment by felipetesc** [comment]
PS: I've used Gravity version 0.5.0.

**Comment by felipetesc** [comment]
Yes, I understood that previously, and I've used the function which the signature is **private func get_public()** from outside the class, inside the main func. Perhaps the previous example is confuse, so here it's another:
```
class Test{
    private func getVal(){
        return "my string";
    }
}

func main(){
    var test = Test();
    var val = test.getVal();// get val is a private method
    System.print(val);
}
```
And the result is: 
my string
RESULT: NULL (in 0.0338 ms)

**Comment by marcobambini** [comment]
@felipetesc seems like an issue that needs to be fixed. Thanks!

**Comment by m0nt3-q** [comment]
hey @marcobambini, i've been looking into this and i think the issue is in `gravity_semacheck2` or `gravity_codegen`, but i'm not sure.

from my debugging, i've found that the `gtoken_t access` for a function with private and public are correct values. however, once they leave `decl_check_access_specifier`, i have no idea where that value is next used.

do you have any idea on what could be happening, and where?
